packages/bvaughn-architecture-demo/components/search-files/HighlightMatch.tsx

```diff
@@ -0,0 +1,24 @@
+import { ReactNode, useMemo } from "react";
+
+import styles from "./HighlightMatch.module.css";
+
+export default function HighlightMatch({ needle, text }: { needle: string; text: string }) {
+  const pieces = useMemo(() => text.split(needle), [needle, text]);
+
+  let children: ReactNode[] = [];
+  for (let index = 0; index < pieces.length; index++) {
+    const piece = pieces[index];
+
+    children.push(<span key={index}>{piece}</span>);
+
+    if (index < pieces.length - 1) {
+      children.push(
+        <span className={styles.Match} key={`${index}-needle`}>
+          {needle}
+        </span>
+      );
+    }
+  }
+
+  return children as any;
```

Comment:
Nit: avoid `any` by using `return <>{children}</>;`

Reply:
This is a goofy aspect of TypeScript IMO. Casting to `any` isn't great, but returning `children as any` is _slightly more performant_ since it avoids creating an unnecessary Fiber.

Reply:
I think that's rather a goofy aspect of the types shipped by React, I wish they would just "admit" that a render function is allowed to return `ReactNode[]`.

Reply:
Agreed that should be the return type. I don't know the legacy reasons for why it is what it is.

FWIW though the React team doesn't ship TypeScript types. They're community maintained ([over at DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts)).